featuretests/cmd_juju_resources_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package featuretests
+
+import (
+	"strings"
+
+	"github.com/juju/cmd/cmdtesting"
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	charmresource "gopkg.in/juju/charm.v6-unstable/resource"
+
+	"github.com/juju/juju/charmstore"
+	"github.com/juju/juju/cmd/juju/resource"
+	coretesting "github.com/juju/juju/juju/testing"
+	"github.com/juju/juju/state"
+)
+
+type ResourcesCmdSuite struct {
+	coretesting.JujuConnSuite
+
+	appOne *state.Application
+
+	charmName   string
+	appOneName  string
+	unitOneName string
+
+	client *stubCharmStore
+}
+
+func (s *ResourcesCmdSuite) SetUpTest(c *gc.C) {
+	s.JujuConnSuite.SetUpTest(c)
+
+	s.charmName = "starsay"
+	s.appOneName = "app1"
+	charmOne := s.AddTestingCharm(c, s.charmName)
+
+	var err error
+	s.appOne, err = s.State.AddApplication(state.AddApplicationArgs{
```

Comment:
Using the factory is preferable here since there's no error checking required - the factory does it. Plus it's the preferred way of creating test entities.